src/browser/of_events.ts

```diff
@@ -21,11 +21,13 @@ class OFEvents extends EventEmitter {
         super();
     }
 
-    public emit(routeString: string, ...data: any[]) {
+    public emit(routeString: string, payload: any, ...extraArgs: any[]) {
         const tokenizedRoute = routeString.split('/');
-
+        const eventPropagations = new Map<string, any>();
+        const data = [payload, ...extraArgs];
```

Comment:
So now `this.emit(...)` and `super.emit(...)` have different call signatures?

Consider instead keeping the original declaration, and defining `payload` and `extraArgs` from `data` instead of vice-versa:

```
public emit(routeString: string, ...data: any[]) {
  // ...
  const [payload, ...extraArgs] = data;
  // ...
```